### PR TITLE
Webhook documentation, CLI commands for listing webhooks

### PR DESCRIPTION
**Pull request by Kami**
Webhook documentation, CLI commands for listing webhooks.

### CODE REVIEW

File: docs/source/webhooks.rst [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+Another example where webhooks come handy is when you want to consume events
+from a 3rd party services which already offer webhooks integration - an example
+of such service includes Github.
+
+Authentication
+--------------
+
+All the requests to the /webhooks endpoints needs to be authenticated in the
+same way as other API requests. This means the requests need to contain
+``X-Auth-Token`` header with a valid authentication token.
```
Comment (jfryman) [inline docs/source/webhooks.rst]: :cool:, but where do I set this?

Comment (Kami) [inline docs/source/webhooks.rst]: You need to pass this value inside the HTTP request header. The way you set it depends on the HTTP client you use (curl, programming language http client, service, etc.).

On a related note - we should allow user to also pass in a token via a query string - this also allow users to use the webhook with services which don't allow you to set custom headers (e.g. Github). I believe this will also help @enykeev

Comment (Kami) [inline docs/source/webhooks.rst]: Will make that change in a separate PR.

Comment (jfryman) [inline docs/source/webhooks.rst]: I'm actually talking on our end? where is `matoken` set in Stackstorm? In the rule?

Comment (Kami) [inline docs/source/webhooks.rst]: Sorry, I'm slightly confused - you mean how to set it inside an action? Or how to set it via the python client (st2client)?

Or you mean how are tokens generated? If so, tokens are generated by st2auth when user authenticates with valid credentials.

Comment (jfryman) [inline docs/source/webhooks.rst]: I didn't realize `st2auth` is in play here. That is not clear at all.
